File: src/main/scala/labs/lab3_strategy/Elf.scala
package labs.lab3_strategy

/**
  * @author Evgeny Borisov
  */
class Elf(val name: String) extends GameCharacter {
  override var hp: Int = getNumberBetween(min = 7, max = 9)
  override var power: Int = getNumberBetween(5, 7)

  override def kick(enemy: GameCharacter): Int = {
    if (enemy.power < power) {
      val damage = enemy.hp
      enemy.hp = 0
      damage

    } else {
      enemy.power -= 1
      0
    }
  }
}

File: src/main/java/my_spring/Elf.java
package my_spring;

import lombok.Data;

/**
 * @author Evgeny Borisov
 */
@Data
public class Elf {
    @InjectRandomInt(min=5,max=20)
    private int power;
    @InjectRandomInt(min=20,max=30)
    private int speed;
}
